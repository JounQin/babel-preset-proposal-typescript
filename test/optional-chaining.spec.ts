import _ from './optional-chaining'

import { execute } from './helpers'

const proposal = 'optional-chaining'

test(proposal, () => {
  expect(() => execute(proposal)).toThrowErrorMatchingSnapshot()
  expect(_()).toBe(undefined)
  expect(() => _({})).toThrowErrorMatchingSnapshot()
  expect(
    _({
      b: [{}],
    }),
  ).toBe(undefined)
  expect(() =>
    _({
      b: [{ c() {} }],
    }),
  ).toThrowErrorMatchingSnapshot()
  expect(
    _({
      b: [{ c: () => ({ d: 'd' }) }],
    }),
  ).toBe('d')
})
