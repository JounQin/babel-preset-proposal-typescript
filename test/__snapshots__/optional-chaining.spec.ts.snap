// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`optional-chaining 1`] = `"Unexpected token ."`;

exports[`optional-chaining 2`] = `"Cannot read property '0' of undefined"`;

exports[`optional-chaining 3`] = `"Cannot read property 'd' of undefined"`;
